### PR TITLE
Add basic RSS feeds for recent entries and events

### LINKED ISSUE #40: Add RSS Feed to the Recent Entries page
Would love to get those new entries as soon as they are submitted!

### PR DESCRIPTION
**Pull request by PaleoCrafter**
This closes #40 by adding RSS feeds both for the most recent entries as well as all entries in an event. Only the 5 most recent entries will be returned, but i chose that number fairly arbitrarily.

I create this as a draft because feedback on what should be included in the feed would be appreciated. @ArgentSun, you requested that feature, what would you like to be included in the title and description of feed items?

### CODE REVIEW

**Comment by ArgentSun** [comment]
Hmm...

The full text of the first question will often be too long for a title, I think, which is unfortunate. You can either truncate it (though I don't know how useful it would be to see something like "_Are the Ashynite magic system and_" in the title) or just go with the entry ID - which is equally uninformative, but it's at least consistent. So maybe let's start with something like this, and see how it looks:

- **Title:** `[Berlin signing] Entry ID #XXXXX`
- **Description:** `{the full text of the entry, formatted}`

Some additional features to consider:

- If the entry is marked as Paraphrased, add that to the title? e.g. `[Event] Entry ID #XXXXX (Paraphrased)`
- If there is an image source, shove that inside `<image>`
- If there is an audio source, maybe add a play button? Not sure how I feel about that.
- Tags and source should probably show up on the bottom too.

**Comment by PaleoCrafter** [comment]
I've implemented Argent's suggestion as far as I could, with this as the result in my reader:
Current RSS feed
I'd like to also add image and potentially audio sources through RSS enclosures, but my local environment currently doesn't seem to correctly display images. @m-strzelczyk, if you've got any hints at all in this regard, any help would be appreciated.

File: palanaeum/templates/palanaeum/feeds/entry_description.html [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{% spaceless %}
+{% load i18n %}
+{% endspaceless %}{% for line in obj.lines %}
```
[the remark this answers is not in the document]
Comment (PaleoCrafter) [reply palanaeum/templates/palanaeum/feeds/entry_description.html]: I wholeheartedly agree with that sentiment. Partially, the indentation was botched by VS Code which was a little overeager with its (arguably weird) formatting upon copy/pasting code. I prefer the indented style as well and will adjust the template appropriately.

File: palanaeum/feeds.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+class RecentEntriesFeed(Feed):
+    title = "%s - Recent Entries" % get_config('page_title')
+    link = "/recent/"
+    title_template = "palanaeum/feeds/entry_title.html"
+    description_template = "palanaeum/feeds/entry_description.html"
+
+    def items(self):
+        entries_ids = Entry.all_visible.order_by('-created').values_list('id', flat=True)[:10]
+        entries_map = Entry.prefetch_entries(entries_ids, show_unapproved=True)
```
[the remark this answers is not in the document]
Comment (PaleoCrafter) [reply palanaeum/feeds.py]: I wasn't quite certain what to do about unapproved entries. Maybe a GET query parameter would be appropriate? Considering this was inspired by Argent's feature request, I'd like to allow staff members to also receive news about *any* new entries using the feed.

File: palanaeum/feeds.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    def get_object(self, request, event_id):
+        event = get_object_or_404(Event, pk=event_id)
+
+        if not event.visible():
+            raise PermissionDenied
+
+        return event
+    
+    def title(self, event):
+        return "%s - %s" % (get_config('page_title'), event.name)
```
[the remark this answers is not in the document]
Comment (PaleoCrafter) [reply palanaeum/feeds.py]: Would you like me to change it to something else? I don't know what Python version the live environment is running, but I'd be partial to f-strings. :P

**Comment by PaleoCrafter** [comment]
I've got images to work in my Docker environment and have added them to the feed:
Feed with images
I've also added a note to the entry title which indicates there's an audio snippet attached to the entry.

File: palanaeum/feeds.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (m-strzelczyk) [reply palanaeum/feeds.py]: Python 3.5.

Nah, no need to change it, I just live my whole life with str.format() ;) I also like the f-string syntax, but that's not that popular yet.

File: palanaeum/feeds.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (m-strzelczyk) [reply palanaeum/feeds.py]: OK, adding an optional GET parameter should be fine. @ArgentSun what do you think? Should the RSS feed serve the unapproved version of entries by default or not? The normal webpage shows unapproved entries to logged in users, but only approved for anonymous ones. Should we treat RSS queries like if they come from someone logged in or not?

[the remark this answers is not in the document]
Comment (ArgentSun) [reply palanaeum/feeds.py]: Hidden entries, suggestions, and unapporved entries should all be excluded, I'd say. RSS feeds are public, and the Arcanists already have a way of keeping track of new stuff that needs approval.

[the remark this answers is not in the document]
Comment (PaleoCrafter) [reply palanaeum/feeds.py]: Great, I've updated the code to not include any unapproved entries.

**Pull request merged**